protostar-rust/src/main.rs

```diff
@@ -1,30 +1,37 @@
-use anyhow::{Context, Result};
+use anyhow::{anyhow, Context, Result};
 use camino::Utf8PathBuf;
 use clap::Parser;
+use include_dir::{include_dir, Dir};
 use scarb_metadata::MetadataCommand;
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
 
 use rust_test_runner::pretty_printing;
 use rust_test_runner::run_test_runner;
 
+static CORELIB: Dir = include_dir!("../cairo/corelib/src");
+
 #[derive(Parser, Debug)]
 struct Args {
     test_filter: Option<String>,
-    // TODO #1997 this is a temporary solution for tests to work, this argument
-    //  should be detected automatically
-    #[arg(short, long)]
-    corelib_path: Option<String>,
+}
+
+fn load_corelib() -> Result<TempDir> {
+    let tmp_dir = tempdir()?;
+    CORELIB
+        .extract(&tmp_dir)
+        .expect("Failed to copy corelib to temporary directory");
+    Ok(tmp_dir)
 }
 
 fn main_execution() -> Result<()> {
-    let args = Args::parse();
+    let _args = Args::parse();
```

Comment:
why is this not removed?

Reply:
Because it is being used in other PR (#1999). The CLI main interface prototype was designed for a specific purpose of handling arguments of the CLI.